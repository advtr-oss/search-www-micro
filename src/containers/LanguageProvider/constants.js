/*
 *
 * LanguageProvider constants
 *
 */

export const CHANGE_LOCALE = 'app/LanguageToggle/CHANGE_LOCALE';

export const LOCALE_KEY = 'locale'
