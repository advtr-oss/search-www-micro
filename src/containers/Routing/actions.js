import s from 'react'

const la = {
  env: Bt.Env.PRODUCTION,
  zoom: 17,
  placeholder: Object(Pe.t)('wazemap.search.placeholder'),
  searchProvider: sa(),
  requestDelayMs: 200,
  theme: Vt.SearchTheme.DEFAULT,
  items: [],
  title: ve.a.translate('livemap.search.suggestions_title'),
  onComplete: e => {},
  onClear: () => {},
  onInputClear: () => {}
}

function ua (e) {
  let t, n
  const { primary: a, secondary: r, suggestion: o, setSuggestion: i } = (e => {
    let t, n
    const [a, r] = s.useState((t = e.primary) !== null && void 0 !== t ? t : '')
    const [o, i] = s.useState((n = e.secondary) !== null && void 0 !== n ? n : '')
    const c = s.useMemo(() => o.replace(a, '').replace(/^[,]/, '').trim(), [a, o])
    const l = s.useMemo(() => [a, o].filter(Boolean).join(', '), [a, o])
    s.useEffect(() => {
      let t, n
      r((t = e.primary) !== null && void 0 !== t ? t : ''),
      i((n = e.secondary) !== null && void 0 !== n ? n : '')
    }
    , [e.primary, e.secondary])
    const u = s.useCallback(e => {
      let t, n
      typeof e === 'string'
        ? r(e)
        : (r((t = e.primary) !== null && void 0 !== t ? t : ''),
          i((n = e.secondary) !== null && void 0 !== n ? n : ''))
    }
    , [])
    return {
      primary: a,
      secondary: c,
      suggestion: l,
      setSuggestion: u
    }
  }
  )(e)
  const [c, l] = s.useState(!o)
  const [m, d] = s.useState(!1)
  const [p, v] = s.useState(a)
  const [E, h] = s.useState(a)
  const { dispatch: f, state: g } = s.useContext(te)
  const { translator: _, browser: w } = s.useContext(pe)
  const b = s.useContext(Rn.a)
  const O = s.useMemo(() => {
    let e
    return ((e = g.user) === null || void 0 === e ? void 0 : e.favorites) == null ? [] : o.length === 0 ? g.user.favorites : g.user.favorites.filter(e => kn(e.label, o))
  }
  , [o, g.user])
  const [y, C] = Bn(O)
  const N = e.withLocation && !(a && y.length === 0)
  const L = s.useMemo(() => {
    const e = Object(Pe.t)('livemap.favorites.home').toLowerCase()
    const t = Object(Pe.t)('livemap.favorites.work').toLowerCase()
    const n = o.trim().toLowerCase()
    return n.length > 0 && (e.startsWith(n) || t.startsWith(n))
  }
  , [o])
  s.useEffect(() => {
    v(e.primary || '')
  }
  , [e.primary]),
  s.useEffect(() => {
    l(!e.primary),
    e.fromToSwitched && C([])
  }
  , [e.primary]),
  s.useEffect(() => {
    let t
    (t = e.onFocusedChange) === null || void 0 === t || t.call(e, m)
  }
  , [m]),
  s.useEffect(() => {
    o || l(!0)
  }
  , [o])
  const S = Object(gt.d)('REQUEST')
  const T = Object(gt.d)('DELETED')
  const R = Object(gt.d)('ITEM_CLICK')
  const k = Object(gt.d)('LOGIN_HOOK_SHOWN')
  const D = Object(gt.d)('LOGIN_HOOK_CLICK')
  const [A, x] = s.useState(Date.now())
  const j = (t = e.searchProvider.name) !== null && void 0 !== t ? t : 'UNKNOWN'
  const P = (function ({ searchProvider: e, env: t, searchCenter: n }) {
    const { browser: a } = Object(s.useContext)(pe)
    const r = Object(s.useContext)(Rn.a)
    return Object(s.useCallback)(o => Fn(this, void 0, void 0, function * () {
      const i = {
        browser: a,
        map: r
      }
      const s = ve.a.locale
      const c = {
        env: t,
        locale: s
      }
      if (e.supportsCenterRadius && n != null) {
        const { center: t, radius: a } = Gn(n, i)
        return e.search(o, Object.assign(Object.assign({}, c), {
          center: t,
          radius: a
        }))
      }
      const { circle: l, viewport: u } = yield Hn(i)
      return e.supportsViewport && u != null
        ? e.search(o, Object.assign(Object.assign({}, c), {
          viewportLo: u.lo,
          viewportHi: u.hi
        }))
        : e.supportsCenterRadius && l != null ? e.search(o, Object.assign(Object.assign({}, c), l)) : e.search(o, c)
    }
    ), [t, n, a, r, e])
  }(e))
  const z = s.useCallback(Object(ue.debounce)(t => ca(this, void 0, void 0, function * () {
    let n
    (n = e.onInput) === null || void 0 === n || n.call(e, t)
    const a = Date.now()
    let r = []
    let o = !1
    const s = E.length === 0 && t.length !== 0
    if (s && x(Date.now()),
    i(t),
    t.length) {
      try {
        r = yield P(t)
      } catch (e) {
        o = !0
      }
      S({
        params: {
          LATENCY_MS: Date.now() - a,
          QUERY: t,
          NUM_RESULTS: r.length,
          NEW_QUERY: s,
          HAS_ERROR: o,
          PROVIDER: j,
          HYBRID_EXP: an(Zt.HYBRID_VIEWPORT)
        }
      })
    } else {
      T({
        params: {
          METHOD: 'INPUT',
          PROVIDER: j,
          HYBRID_EXP: an(Zt.HYBRID_VIEWPORT)
        }
      })
    }
    h(t),
    C(r)
  }
  ), e.requestDelayMs, e.requestDelayMs === 0), [P, T, S, i])
  const M = s.useCallback(() => {
    d(!0)
  }
  , [d])
  const U = s.useRef()
  const H = s.useCallback(t => {
    let n
    t && i({
      primary: t,
      secondary: r
    }),
    (n = e.onBlur) === null || void 0 === n || n.call(e),
    U.current = setTimeout(() => {
      h(t),
      l(!t),
      d(!1),
      C([])
    }
    , 200)
  }
  , [r, i])
  const G = s.useCallback((t, n, a, r) => ca(this, void 0, void 0, function * () {
    let s, c, m, p, g, w
    if (t.type == u.h.LOGIN_HOOK) {
      return D({}),
      void On(f, _.locale)
    }
    f(new I(!0)),
    U.current && clearTimeout(U.current)
    const O = 'place' in t
      ? t.place
      : yield e.searchProvider.getDetails({
        label: t.label,
        value: t.value
      })
    const y = t.type != u.h.FAVORITE || t.favoriteType != u.c.HOME && t.favoriteType != u.c.WORK ? t.type : t.favoriteType
    const C = `${O.name}, ${O.address}`
    i({
      primary: O.name,
      secondary: O.address
    }),
    v(O.name || ''),
    h(O.name || ''),
    l(!1),
    d(!1)
    const N = Date.now() - A
    R({
      params: {
        RESULT_TYPE: y,
        LON: (s = O.latLng) === null || void 0 === s ? void 0 : s.lng,
        LAT: (c = O.latLng) === null || void 0 === c ? void 0 : c.lat,
        LINE_NUMBER: a,
        NUM_RESULTS: r,
        QUERY: E,
        DISTANCE: b != null ? new Re.LatLng(O.latLng.lat, O.latLng.lng).distanceTo(b.getCenter()) : void 0,
        GOOGLE_PLACE_ID: (m = O.rawGooglePlaceData) === null || void 0 === m ? void 0 : m.place_id,
        ADDRESS: (p = O.rawGooglePlaceData) === null || void 0 === p ? void 0 : p.formatted_address,
        GOOGLE_PLACE_TYPES: (g = O.rawGooglePlaceData) === null || void 0 === g ? void 0 : g.types.join(','),
        IS_ESTABLISHMENT: (w = O.rawGooglePlaceData) === null || void 0 === w ? void 0 : w.types.includes(u.g.ESTABLISHMENT),
        QUERY_MATCHES_LOCALIZED_HOME_OR_WORK: L,
        PROVIDER: j,
        QF_TIME: N,
        HYBRID_EXP: an(Zt.HYBRID_VIEWPORT)
      }
    })
    const S = y === u.h.SUGGESTION
      ? {
          index: n,
          suggestionIndex: a != null ? a : 0,
          queryTimeMs: N,
          queryText: o,
          text: C,
          place: O,
          type: y
        }
      : {
          index: n,
          text: t.label,
          place: O,
          type: y
        }
    e.onComplete(S)
  }
  ), [e, R, o, b, L, j, D, f, _.locale, A, i, E])
  const F = s.useCallback(() => {
    i(''),
    h(''),
    C([]),
    e.onClear(),
    T({
      params: {
        METHOD: 'CLEAR'
      }
    })
  }
  , [e.onClear, T, i])
  const B = s.useCallback(() => {
    i(''),
    v(''),
    h(''),
    C([]),
    e.onInputClear && e.onInputClear(),
    T({
      params: {
        METHOD: 'CLEAR'
      }
    }),
    U.current && clearTimeout(U.current)
  }
  , [e.onInputClear, T, i])
  const V = s.useRef(null)
  const W = s.useCallback(e => {
    const t = Object(Wt.parseLonLat)(e.value)
    if (t != null) {
      const e = `lat=${t.lat} lng=${t.lng}`
      const n = {
        label: e,
        place: {
          latLng: t,
          name: e
        }
      }
      G(Object.assign(Object.assign({}, n), {
        type: u.h.RAW
      }), 0, 1)
    }
  }
  , [])
  const K = s.useCallback(t => {
    t.key === 'Escape' && (d(!1),
    i({
      primary: e.primary,
      secondary: e.secondary
    })),
    V.current ? V.current.onKeyDown(t) : t.key === 'Enter' && W(t.target)
  }
  , [e.primary, e.secondary])
  const $ = !w.isMobile && (g.user == null || ((n = g.user) === null || void 0 === n ? void 0 : n.id) === -1) && o.length == 0
  const Y = c && m && (O.length > 0 || N || $ || y.length > 0)
  return s.useEffect(() => {
    $ && Y && k({})
  }
  , [$, Y, k]),
  s.createElement('div', {
    className: re('wm-search', 'theme-' + e.theme, c ? 'is-editable' : 'is-read-only', o ? 'has-value' : 'is-blank'),
    onKeyDown: K
  }, c
    ? s.createElement(Sn, {
      placeholder: e.placeholder,
      value: a,
      selection: p,
      isRouting: e.isRouting,
      onInput: z,
      onFocus: M,
      onBlur: H,
      onClear: B
    })
    : s.createElement(Tn, {
      primary: a,
      secondary: r,
      poiUrl: e.poiUrl,
      isRouting: e.isRouting,
      onSelect: () => l(!0),
      onClear: F
    }), Y && s.createElement(In, {
    container: e.suggestionsContainer
  }, s.createElement(Ln, {
    ref: V,
    items: y,
    selection: a,
    onSelect: G,
    withGeolocation: N,
    favorites: O,
    withFavoritesLoginHook: $,
    suggestionMatchesHomeOrWork: L,
    title: e.title
  })))
}
