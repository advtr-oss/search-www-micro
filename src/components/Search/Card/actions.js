import { SET_SELECTED, CLEAR_SELECTED } from './constants'

export function setSelected (place) {
  return {
    type: SET_SELECTED,
    // This may just set the placeid & analytics elements in the future
    selected: { ...place, loading: false }
  }
}

export function clearSelected () {
  return {
    type: CLEAR_SELECTED
  }
}
