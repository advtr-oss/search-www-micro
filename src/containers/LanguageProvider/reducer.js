/*
 *
 * LanguageProvider reducer
 *
 * Now actively looks for the users language
 *
 * @note: Not the best but will work
 */

import produce from 'immer'
import get from 'lodash.get'
import split from 'lodash.split'
import includes from 'lodash.includes'

import { CHANGE_LOCALE, LOCALE_KEY } from './constants'
import { DEFAULT_LOCALE, appLocales } from '../../i18n'

// Detect user language.
const userLanguage =
  window.localStorage.getItem(LOCALE_KEY) ||
  window.navigator.language ||
  window.navigator.userLanguage

let foundLanguage = includes(appLocales, userLanguage) && userLanguage

if (!foundLanguage) {
  // Split user language in a correct format.
  const userLanguageShort = get(split(userLanguage, '-'), '0')

  // Check that the language is included in the admin configuration.
  foundLanguage = includes(appLocales, userLanguageShort) && userLanguageShort
}

export const initialState = {
  locale: foundLanguage || DEFAULT_LOCALE
}

/* eslint-disable default-case, no-param-reassign */
const languageProviderReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_LOCALE:
        window.localStorage && window.localStorage.setItem(LOCALE_KEY, action.locale)
        draft.locale = action.locale
        break
    }
  })

export default languageProviderReducer
