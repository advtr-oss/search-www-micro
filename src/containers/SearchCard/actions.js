/*
 *
 * SearchCard actions
 *
 */

import { CLEARED_CARD, SELECTED_POI } from './constants'

export function selectPOI (selected) {
  return {
    type: SELECTED_POI,
    selected: { ...selected, loading: false }
  }
}

export function clearPOI () {
  return {
    type: CLEARED_CARD
  }
}
