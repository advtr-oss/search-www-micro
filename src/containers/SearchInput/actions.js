/**
 * SearchInput action
 * */

import { UPDATE_SEARCH, CLEAR_SEARCH } from './constants'

export function updateSearch (value) {
  return {
    type: UPDATE_SEARCH,
    value
  }
}

export function clearSearch () {
  return {
    type: CLEAR_SEARCH
  }
}
