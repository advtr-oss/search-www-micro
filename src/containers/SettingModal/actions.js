/*
 *
 * LanguageProvider actions
 *
 */

import { OPEN_MODAL, CLOSE_MODAL } from './constants'

export function openModal () {
  return {
    type: OPEN_MODAL,
    isOpen: true
  }
}

export function closeModal () {
  return {
    type: CLOSE_MODAL,
    isOpen: false
  }
}
