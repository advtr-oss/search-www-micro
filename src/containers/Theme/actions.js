/*
 *
 * Theme actions
 *
 */

import { DARK_MODE } from './constants'

export function changeTheme (darkMode) {
  return {
    type: DARK_MODE,
    darkMode
  }
}
