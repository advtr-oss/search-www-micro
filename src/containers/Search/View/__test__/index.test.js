/**
 * @jest-environment jsdom
 */

import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { GlobalStyle, ThemeProvider } from '@advtr/tidy'
import LanguageProvider from '../../../../providers/LanguageProvider'

import { translationMessages } from '../../../../i18n'

import configureStore from '../../../../configureStore'
import { View } from '../index'
import { changeLocale } from '../../../../providers/LanguageProvider/actions'

const store = configureStore()

// Add theme provider here too, will look at moving this out and adding a generic one
const redux = (ui, opts) => render((
  <Provider store={store}>
    <ThemeProvider>
      <LanguageProvider messages={translationMessages}>
        <React.Fragment>
          <GlobalStyle />
          <React.Fragment>{ui}</React.Fragment>
        </React.Fragment>
      </LanguageProvider>
    </ThemeProvider>
  </Provider>
), opts)

describe('Search.View', function () {
  it('should render a snapshot', function () {
    const { container } = redux(<View />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with changed locale', function () {
    store.dispatch(changeLocale('fr'))

    const { container } = redux(<View />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
