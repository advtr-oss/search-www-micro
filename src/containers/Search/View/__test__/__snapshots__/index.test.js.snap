// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.View should render a snapshot 1`] = `
.c6 {
  width: 100%;
  border-style: none;
  color: #24292f;
  background-color: #ffffff;
}

.c6::-webkit-input-placeholder {
  color: #6e7781;
}

.c6::-moz-placeholder {
  color: #6e7781;
}

.c6:-ms-input-placeholder {
  color: #6e7781;
}

.c6::placeholder {
  color: #6e7781;
}

.c6:disabled {
  color: #57606a;
  background-color: #f6f8fa;
}

.c6:focus {
  outline: none;
}

.c3 {
  position: relative;
}

.c3 {
  background-color: #ffffff !important;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  padding: 0 8px;
  border-radius: 4px;
  width: 100%;
  height: 48px;
}

.c3:disabled {
  color: #57606a !important;
  background-color: #f6f8fa !important;
}

.c3 > .c5 {
  padding: 0 8px 0 8px;
  color: inherit;
  background-color: inherit;
  line-height: 40px;
  font-weight: 500;
}

.c4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 8px;
}

.c4 {
  border-style: none;
  border-radius: 10px;
  height: 40px;
}

.c4 {
  color: #24292f;
  background-color: inherit;
}

.c4:disabled {
  cursor: default;
}

.c4:disabled > svg {
  cursor: default;
  color: #57606a;
}

.c4 > svg {
  width: 16px;
  height: 16px;
}

.c4:enabled > svg:hover {
  color: #0969da;
}

.c0 {
  position: relative;
}

.c1 {
  display: grid;
  gap: 0;
}

.c1:focus-within > .search--input > div {
  border-radius: 0;
  border-bottom: 1px solid #d0d7de;
  box-shadow: none !important;
}

.c1:focus-within > .search--dropdown > ul {
  border-radius: 0 0 4px 4px;
  box-shadow: 0 3px 6px -6px rgba(140,149,159,0.15) !important;
}

.c2 button {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.c2 > div {
  border-radius: 0 0 4px 4px;
}

@media (min-width:544px) {
  .c1 {
    gap: 16px 0;
  }

  .c1 .search--input {
    border-radius: 4px;
  }
}

@media (min-width:544px) {
  .c1:focus-within > .search--input > div {
    border-radius: 4px;
    border-bottom: none;
    box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
  }
}

@media (min-width:544px) {
  .c1:focus-within > .search--dropdown > ul {
    border-radius: 4px;
    box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
  }
}

@media (min-width:544px) {
  .c2 > div {
    border-radius: 4px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 search--input"
    >
      <div
        class="c3"
      >
        <button
          class="c4"
        >
          <svg
            aria-label="input-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </button>
        <input
          aria-disabled="false"
          aria-label="Search Autocomplete"
          aria-readonly="false"
          autocomplete="off"
          class="c5 c6 input--empty"
          placeholder="Search the world..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Search.View should render with changed locale 1`] = `
.c6 {
  width: 100%;
  border-style: none;
  color: #24292f;
  background-color: #ffffff;
}

.c6::-webkit-input-placeholder {
  color: #6e7781;
}

.c6::-moz-placeholder {
  color: #6e7781;
}

.c6:-ms-input-placeholder {
  color: #6e7781;
}

.c6::placeholder {
  color: #6e7781;
}

.c6:disabled {
  color: #57606a;
  background-color: #f6f8fa;
}

.c6:focus {
  outline: none;
}

.c3 {
  position: relative;
}

.c3 {
  background-color: #ffffff !important;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  padding: 0 8px;
  border-radius: 4px;
  width: 100%;
  height: 48px;
}

.c3:disabled {
  color: #57606a !important;
  background-color: #f6f8fa !important;
}

.c3 > .c5 {
  padding: 0 8px 0 8px;
  color: inherit;
  background-color: inherit;
  line-height: 40px;
  font-weight: 500;
}

.c4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 8px;
}

.c4 {
  border-style: none;
  border-radius: 10px;
  height: 40px;
}

.c4 {
  color: #24292f;
  background-color: inherit;
}

.c4:disabled {
  cursor: default;
}

.c4:disabled > svg {
  cursor: default;
  color: #57606a;
}

.c4 > svg {
  width: 16px;
  height: 16px;
}

.c4:enabled > svg:hover {
  color: #0969da;
}

.c0 {
  position: relative;
}

.c1 {
  display: grid;
  gap: 0;
}

.c1:focus-within > .search--input > div {
  border-radius: 0;
  border-bottom: 1px solid #d0d7de;
  box-shadow: none !important;
}

.c1:focus-within > .search--dropdown > ul {
  border-radius: 0 0 4px 4px;
  box-shadow: 0 3px 6px -6px rgba(140,149,159,0.15) !important;
}

.c2 button {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.c2 > div {
  border-radius: 0 0 4px 4px;
}

@media (min-width:544px) {
  .c1 {
    gap: 16px 0;
  }

  .c1 .search--input {
    border-radius: 4px;
  }
}

@media (min-width:544px) {
  .c1:focus-within > .search--input > div {
    border-radius: 4px;
    border-bottom: none;
    box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
  }
}

@media (min-width:544px) {
  .c1:focus-within > .search--dropdown > ul {
    border-radius: 4px;
    box-shadow: 0 3px 6px rgba(140,149,159,0.15) !important;
  }
}

@media (min-width:544px) {
  .c2 > div {
    border-radius: 4px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 search--input"
    >
      <div
        class="c3"
      >
        <button
          class="c4"
        >
          <svg
            aria-label="input-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </button>
        <input
          aria-disabled="false"
          aria-label="Recherch Remplissage automatique"
          aria-readonly="false"
          autocomplete="off"
          class="c5 c6 input--empty"
          placeholder="Cherchez le monde..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
